cpp/src/arrow/acero/util_test.cc

```diff
@@ -184,5 +185,117 @@ TEST(FieldMap, ExtensionTypeHashJoin) {
   EXPECT_EQ(i.get(0), 0);
 }
 
+template <typename Queue>
+void ConcurrentQueueBasicTest(Queue& queue) {
+  ASSERT_TRUE(queue.Empty());
```

Comment:
```suggestion
#ifndef ARROW_ENABLE_THREADING
  GTEST_SKIP() << "Test requires threading enabled";
#endif
  ASSERT_TRUE(queue.Empty());
```

Reply:
Arrow can be built w/o threading support, so every test using explicit threading should skip like this.